pypeit/edgetrace.py

```diff
@@ -3572,6 +3584,43 @@ def peak_refine(self, rebuild_pca=False, debug=False):
         if ntrace < self.ntrace:
             msgs.warn('Found fewer traces using peak finding than originally available.  '
                       'May want to reset peak threshold.')
+            
+        if self.par['trace_rms_tol'] is not None:
+            # Get the PCA reference row.  The PCA *must* have been defined to get
+            # this far (see pcatype test at the beginning of the function).
+            reference_row = self.left_pca.reference_row if self.par['left_right_pca'] \
+                                else self.pca.reference_row
+            # Match the new trace positions to the input ones
+            gpm = np.logical_not(self.bitmask.flagged(self.edge_msk[reference_row]))
+            # TODO: Include a tolerance here?  Needs testing with more datasets.
+            peak_indx = slitdesign_matching.match_positions_1D(
+                            self.edge_fit[reference_row][gpm],
+                            fit[reference_row])
+
+            # Determine the RMS difference between the input and output traces.
+            # This allows us to compare traces that had already been identified
+            # to their new measurements resulting from peak_trace, and remove
+            # them if they are too discrepant from their original form.  This is
+            # largely meant to find and remove poorly constrained traces, where
+            # the polynomial fit goes wonky.
+            diff = fit - self.edge_fit.T[gpm][peak_indx].T
+            rms = np.sqrt(np.mean((diff - np.mean(diff, axis=0)[None,:])**2, axis=0))
+
+            # TODO: Add a report to the screen or a QA plot?
+
+            # Select traces below the RMS tolerance or that were newly
+            # identified by peak_trace.  I.e., this will *not* catch newly
+            # identified traces found by peak_trace that are also poorly
+            # constrained!
+            indx = (rms < self.par['trace_rms_tol']) | (peak_indx == -1)
+            if not all(indx):
```

Comment:
`if not np.all(indx):` ?

Reply:
`all` actually works, but, yes, good call.